metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -2377,11 +2398,17 @@ class MetalsLanguageServer(
       workspaceSymbols.indexClasspath()
     }
     timerProvider.timedThunk(
-      "indexed workspace SemanticDBs",
+      "indexed workspace Scala SemanticDBs",
       clientConfig.initialConfig.statistics.isIndex
     ) {
       semanticDBIndexer.onScalacOptions(i.scalacOptions)
     }
+    timerProvider.timedThunk(
+      "indexed workspace Java SemanticDBs",
+      clientConfig.initialConfig.statistics.isIndex
+    ) {
+      semanticDBIndexer.onJavacOptions(i.javacOptions)
```

Comment:
@tgodzik It's likely that projects with Scala and Java files will have the same TargetRoot set under `scalacOptions` and `javacOptions`.  So calling `semanticDBIndexed.onScalacOptions` and `semanticDBIndexed.onJavacOptions` will be doing the same thing so `SemanticDBIndexer#onChangeDirectory` will be called twice for each target.  Is that a performance issue?

Reply:
That might be, so it would be cool to avoid that if it's the same directory.